From: Ilya Dryomov <ilya.dryomov@inktank.com>
Date: Tue, 28 Jan 2014 19:16:18 +0200
Subject: [PATCH] ceph: cast PAGE_SIZE to size_t in ceph_sync_write()
Git-commit: 125d725c923527a85876c031028c7f55c28b74b3
Patch-mainline: v3.14
References: Fate#318586


Use min_t(size_t, ...) instead of plain min(), which does strict type
checking, to avoid compile warning on i386.

Cc: Jianpeng Ma <majianpeng@gmail.com>
Signed-off-by: Ilya Dryomov <ilya.dryomov@inktank.com>
Reviewed-by: Sage Weil <sage@inktank.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/file.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/ceph/file.c
+++ b/fs/ceph/file.c
@@ -765,7 +765,7 @@ static ssize_t ceph_sync_write(struct ki
 
 		left = len;
 		for (n = 0; n < num_pages; n++) {
-			size_t plen = min(left, PAGE_SIZE);
+			size_t plen = min_t(size_t, left, PAGE_SIZE);
 			ret = iov_iter_copy_from_user(pages[n], &i, 0, plen);
 			if (ret != plen) {
 				ret = -EFAULT;
